java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -1038,6 +1040,36 @@ public Description matchVariable(VariableTree tree, VisitorState state) {
     return Description.NO_MATCH;
   }
 
+  // This method removes unnecessary blocks from lambda expression bodies.
+  // This scenario is possible in case of single return statements and expression statements
+  // in the lambda body after Piranha removes the code related to stale flags.
+  @Override
+  public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState visitorState) {
+    Tree lambdaBody = tree.getBody();
+    if (lambdaBody instanceof BlockTree) {
+      BlockTree bt = (BlockTree) lambdaBody;
+      // Block should contain only one statement
+      if (bt.getStatements().size() == 1) {
+        StatementTree singleStmt = bt.getStatements().get(0);
+        ExpressionTree expr = null;
+        if (singleStmt instanceof ReturnTree) {
+          ReturnTree stmt = (ReturnTree) singleStmt;
+          expr = stmt.getExpression();
+        }
+        if (singleStmt instanceof ExpressionStatementTree) {
+          ExpressionStatementTree expStmt = (ExpressionStatementTree) singleStmt;
+          expr = expStmt.getExpression();
+        }
+        if (expr != null) {
+          return buildDescription(tree)
+              .addFix(SuggestedFix.replace(bt, visitorState.getSourceForNode(expr)))
```

Comment:
Might be safer to build a new lambda expression with `expr` as the body and replace `tree` (but again, this is moot, since this won't do the rewrite you want).